cmake/TaichiCore.cmake

```diff
@@ -62,12 +62,15 @@ if (TI_WITH_OPENGL)
     target_include_directories(${LIBRARY_NAME} PUBLIC ${GLEW_INCLUDE_DIRS})
     target_link_libraries(${LIBRARY_NAME} ${GLEW_LIBRARIES} GLEW)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGLEW_STATIC")
-    find_package(glfw3 REQUIRED)
-    if (NOT glfw3_FOUND)
-      message(FATAL_ERROR "glfw3 not found.")
-    endif()
-    message("Building with glfw ${glfw3_VERSION}")
-    target_link_libraries(${LIBRARY_NAME} glfw)
+
+    set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "Build the GLFW example programs")
+    set(GLFW_BUILD_TESTS OFF CACHE BOOL "Build the GLFW test programs")
+    set(GLFW_BUILD_DOCS OFF CACHE BOOL "Build the GLFW documentation")
+    set(GLFW_INSTALL OFF CACHE BOOL "Generate installation target")
+
+    message("Building with GLFW")
+    add_subdirectory(external/glfw)
+    target_link_libraries(${LIBRARY_NAME} glfw3)
```

Comment:
Will this add the `taichi/external/glfw/libglfw.a` instead of the `/usr/lib/libglfw.so`?

Reply:
Good question. I just now updated according to https://www.glfw.org/docs/3.3/build_guide.html#build_link_cmake_source.